pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -317,46 +322,77 @@ private void setupStateStore() throws Exception {
         }
     }
 
-    private void processResult(Record srcRecord,
-                               CompletableFuture<JavaExecutionResult> result) throws Exception {
-        result.whenComplete((result1, throwable) -> {
-            if (throwable != null || result1.getUserException() != null) {
-                Throwable t = throwable != null ? throwable : result1.getUserException();
-                log.warn("Encountered exception when processing message {}",
-                        srcRecord, t);
-                stats.incrUserExceptions(t);
-                srcRecord.fail();
-            } else {
-                if (result1.getResult() != null) {
-                    sendOutputMessage(srcRecord, result1.getResult());
+    private void processResult(@SuppressWarnings("rawtypes") Record srcRecord,
+                               JavaExecutionResult result) throws SinkException {
+    	
+    	if (result.getUserException() != null) {
```

Comment:
This is where we do the error handling for the synchronous function calls. The logic inside this block is identical to error handling logic for asynchronous function calls.